helm/hpcc/templates/_helpers.tpl

```diff
@@ -1548,3 +1554,82 @@ A template to output a merged environment. Pass in a list with global then local
   value: {{ $value }}
 {{ end -}}
 {{- end -}}
+
+
+{{/*
+A template to convert a human readable K8s memory string to bytes
+Pass in value
+*/}}
+{{- define "hpcc.k8sMemoryStringToBytes" -}}
+{{- $ctx := dict -}}
+{{- if hasSuffix "i" .value -}}
+ {{- if hasSuffix "Ki" .value -}}
+  {{- $_ := set $ctx "scale" 1024 -}}
+ {{- else if hasSuffix "Mi" .value -}}
+  {{- $_ := set $ctx "scale" 1048576 -}}
+ {{- else if hasSuffix "Gi" .value -}}
+  {{- $_ := set $ctx "scale" 1073741824 -}}
+ {{- else if hasSuffix "Ti" .value -}}
+  {{- $_ := set $ctx "scale" 1099511627776 -}}
+ {{- else if hasSuffix "Pi" .value -}}
+  {{- $_ := set $ctx "scale" 1125899906842624 -}}
+ {{- else if hasSuffix "Ei" .value -}}
+  {{- $_ := set $ctx "scale" 1152921504606846976 -}}
+ {{- else -}}
+  {{- $_ := fail (printf "Invalid size suffix on memory resoure specification: %s" .value) -}}
+ {{- end -}}
+ {{- $_ := set $ctx "number" (substr 0 (int (sub (len .value) 2)) .value) -}}
+{{- else -}}
+ {{- if hasSuffix "K" .value -}}
+  {{- $_ := set $ctx "scale" 100 -}}
+ {{- else if hasSuffix "M" .value -}}
+  {{- $_ := set $ctx "scale" 1000000 -}}
+ {{- else if hasSuffix "G" .value -}}
+  {{- $_ := set $ctx "scale" 1000000000 -}}
+ {{- else if hasSuffix "T" .value -}}
+  {{- $_ := set $ctx "scale" 1000000000000 -}}
+ {{- else if hasSuffix "P" .value -}}
+  {{- $_ := set $ctx "scale" 1000000000000000 -}}
+ {{- else if hasSuffix "E" .value -}}
+  {{- $_ := set $ctx "scale" 1000000000000000000 -}}
+ {{- else -}}
+  {{- $_ := fail (printf "Invalid size suffix on memory resoure specification: %s" .value) -}}
+ {{- end -}}
+ {{- $_ := set $ctx "number" (substr 0 (sub (len .value) 1) .value) -}}
+{{- end -}}
+{{- printf "%d" (mul $ctx.number $ctx.scale) -}}
+{{- end -}}
+
+{{/*
+A template to convert a human readable K8s cpu string to milli cpu units
+Pass in value
+*/}}
+{{- define "hpcc.k8sCPUStringToMilliCPU" -}}
+{{- if hasSuffix "m" .value -}}
+ {{- $number := (substr 0 (int (sub (len .value) 1)) .value) -}}
+ {{- printf "%d" (int $number) -}}
+{{- else -}}
+ {{- printf "%d" (int (mulf (float64 .value) 1000.0)) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+A template to convert bytes into a k8s human friendly string
+Pass in value
+*/}}
+{{- define "hpcc.bytesToK8sMemoryString" -}}
+{{- /* NB: Rounds down to units that are a 10th of unit that values is larger than */ -}}
```

Comment:
Does it? How? 1000th maybe...

Reply:
yes, meant a 1000th will change.
e.g. 2Gi => 2000Mi, 2500Mi => 2500Mi.
It was so I didn't have to deal with fractional cases